azkaban-common/src/main/java/azkaban/executor/ExecutorApiClient.java

```diff
@@ -31,6 +51,63 @@
 @Singleton
 public class ExecutorApiClient extends RestfulApiClient<String> {
 
+  final private static Logger logger = Logger.getLogger(ExecutorApiClient.class);
+
+  final private String DEFAULT_TRUSTSTORE_PATH = "keystore";
+  final private String DEFAULT_TRUSTSTORE_PASSWORD = "changeit";
+  final private boolean isTlsEnabled;
+  final private String truststorePath;
+  final private String truststorePassword;
+  private SSLConnectionSocketFactory tlsSocketFactory;
+
+  @Inject
+  public ExecutorApiClient(final Props azkProps) {
+    this.isTlsEnabled = azkProps.getBoolean(EXECUTOR_CONNECTION_TLS_ENABLED, false);
+    this.truststorePath = azkProps
+        .getString(Constants.JETTY_TRUSTSTORE_PATH, this.DEFAULT_TRUSTSTORE_PATH);
+    this.truststorePassword = azkProps.getString(Constants.JETTY_TRUSTSTORE_PASSWORD,
```

Comment:
Truststore password is already available in private properties. Property name for it is azkaban.csr.truststore.password. Will you be able to reuse it here?

Reply:
`azkaban.csr.truststore.password ` and `azkaban.csr.truststore.location` are used within the mp az-grestin-client for bare-metal executor's interaction with Datavault.  (The `csr` in the name stands for Certificate Signing Request).
I would prefer not to overload these properties for the purpose of the ApiClient.

1. The properties are not very meaningful on the webserver (which shouldn't be fetching certificates from grestin) 
2. It's quite possible (and likely) that we use different keys and certificates for communicating with flow-containers vs the grestin.

Please let me know what you think.